Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeFileConverter.cc

```diff
@@ -3,38 +3,37 @@
 
 #include "Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeFileConverter.h"
 #include "FWCore/Framework/interface/LuminosityBlock.h"
-
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
-#include "CondFormats/Common/interface/FileBlobCollection.h"
 
 #include <memory>
 #include <fstream>
 
 MillePedeFileConverter::MillePedeFileConverter(const edm::ParameterSet& iConfig)
     : inputDir_(iConfig.getParameter<std::string>("fileDir")),
       inputFileName_(iConfig.getParameter<std::string>("inputBinaryFile")),
-      fileBlobLabel_(iConfig.getParameter<std::string>("fileBlobLabel")) {
-  // We define what this producer produces: A FileBlobCollection
-  produces<FileBlobCollection, edm::Transition::EndLuminosityBlock>(fileBlobLabel_);
-}
+      fileBlobLabel_(iConfig.getParameter<std::string>("fileBlobLabel")),
+      putToken_{produces<FileBlobCollection, edm::Transition::EndLuminosityBlock>(fileBlobLabel_)} {}
 
 MillePedeFileConverter::~MillePedeFileConverter() {}
 
-void MillePedeFileConverter::endLuminosityBlockProduce(edm::LuminosityBlock& iLumi, const edm::EventSetup& iSetup) {
+void MillePedeFileConverter::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {}
+
+void MillePedeFileConverter::globalEndLuminosityBlockProduce(edm::LuminosityBlock& iLumi,
+                                                             const edm::EventSetup& iSetup) const {
   edm::LogInfo("MillePedeFileActions") << "Inserting all data from file " << inputDir_ + inputFileName_
                                        << " as a FileBlob to the lumi, using label \"" << fileBlobLabel_ << "\".";
   // Preparing the FileBlobCollection:
-  auto fileBlobCollection = std::make_unique<FileBlobCollection>();
+  FileBlobCollection fileBlobCollection;
 
   // Creating the FileBlob:
   // (The FileBlob will signal problems with the file itself.)
   FileBlob fileBlob{inputDir_ + inputFileName_, true};
```

Comment:
I think this is a problem. WIth concurrent Lumis, you could have the file being written to at the same time it is being read. What is needed is a unique file name for each Lumi. If the file is being created in this job then it would be better to have the module which makes the file put the filename into the LuminosityBlock once the file is closed and then this module would read that file name.

Reply:
thanks for the suggestion. I'll have a look to that.